.github/workflows/terraform.yml

```
@@ -124,9 +128,22 @@ jobs:
 
       - name: Terraform apply
         uses: dflook/terraform-apply@8a1bf89017460503cf302632cb15e8583afa7ac4 # v1.42.0
-        if: github.event_name == 'push'
+        if: github.event_name == 'push' || inputs.terraform_operation == 'apply'
         with:
           label: ${{ vars.RESOURCE_NAME }} ${{ inputs.terraform_root_path }}
+          path: ${{ inputs.terraform_root_path }}
+          auto_approve: ${{ github.event_name == 'workflow_dispatch' }}
```

```suggestion
          auto_approve: ${{ inputs.terraform_operation == 'apply' }}
```

so you are sure that will be true only when requesting for apply